scheduler/scheduler.go

```diff
@@ -197,23 +199,26 @@ func createResources(toCreate chan *ScheduledResource, created chan string) {
 				}
 			}
 			created <- r.Key()
-		}(r)
+			//Release semaphor
+			<-ccLimiter
+		}(r, created, ccLimiter)
 	}
 }
 
 func Create(depGraph DependencyGraph, concurrency int) {
 
 	depCount := len(depGraph)
 
-	channelSize := depCount
-	if concurrency > 0 && concurrency < depCount {
-		channelSize = concurrency
+	concurrencyLimiterLen := depCount
+	if concurrency > 0 && concurrency < concurrencyLimiterLen {
+		concurrencyLimiterLen = concurrency
 	}
 
-	toCreate := make(chan *ScheduledResource, channelSize)
+	ccLimiter := make(chan struct{}, concurrencyLimiterLen)
```

Comment:
One small nitpick: I think you could use `chan interface{}` instead of `chan stuct{}` for a semaphor. I think it's more efficient, but probably it won't make any real difference here.